src/main/java/org/jenkinsci/plugins/workflow/steps/EnvStep.java

```diff
@@ -150,11 +153,37 @@
                     }
                 }
                 return b.toString();
+            } else if (overrides instanceof Map) {
```

Comment:
BTW I lobbied for `ArgumentsAction[Impl]` to persist only the internal, resolved form of arguments, but the design we went with persists the surface form, necessitating extra work like this.

Reply:
@jglick  Okay, is there some action to be taken about that now?

Reply:
Not easily. Just noting some background.